### ISSUE TITLE
why use address(0) as native token?

### ISSUE THREAD

**Issue by armatrix** [Uniswap/v4-core#919]
### Component

General design optimization (improving efficiency, cleanliness, or developer experience)

### Describe the suggested feature and problem it solves.

In solidity, zero value is usually used to handle mint, burn and other scenarios. In many languages, zero value has a special meaning. Using zero value as native token will cause some ambiguity in the check. For example, we usually judge that the address passed in cannot be zero address. Maybe address(F) would be better?

### Describe the desired implementation.

replace native token wrapper from address(0) to address(F).

**Comment by wjmelements** [comment]
> replace native token wrapper from address(0) to address(F).

F is highly likely to become the address of a precompile in the future. 0 is not.

**Comment by floor-licker** [comment]
There's two answers,
1) As @wjmelements mentioned, there's chance of a collision with an actual wallet
2) Using `address(0)` for the native token is common practice because it's already understood as a placeholder for "nothing" or "no specific address" in Ethereum. It keeps things simple and consistent as most developers already use it this way in their code.

**Comment by saucepoint** [comment]
i agree with the comments. using an address other than address(0) will likely have trade offs
